File: src/main/java/land/leets/domain/comment/usecase/CreateComment.java
package land.leets.domain.comment.usecase;

import land.leets.domain.auth.AuthDetails;
import land.leets.domain.comment.presentation.dto.CommentRequest;
import land.leets.domain.comment.presentation.dto.CommentResponse;

public interface CreateComment {
    CommentResponse execute(AuthDetails authDetails, CommentRequest request);
}

File: src/main/kotlin/land/leets/domain/comment/usecase/CreateComment.kt
package land.leets.domain.comment.usecase

import land.leets.domain.auth.AuthDetails
import land.leets.domain.comment.presentation.dto.CommentRequest
import land.leets.domain.comment.presentation.dto.CommentResponse

interface CreateComment {
    fun execute(authDetails: AuthDetails, request: CommentRequest): CommentResponse
}
